### PR TITLE
Resume Generator: Education CRUD

### PR DESCRIPTION
**Pull request by ojusharma**
> [!IMPORTANT]
> Merge #291 and #290 before merging this PR as this branch has been rebased off ojus/resume-workep-crud
## 📝 Description

This PR implements a complete Education CRUD service for managing user educational history as part of the resume generation feature.

**Key changes:**
- Added educationservice with full CRUD operations (Create, Read, Update, Delete)
- Added comprehensive validation logic for:
  - Required fields (institution, degree)
  - Date validation (end_date cannot be before start_date)
  - GPA range validation (0.0 - 5.0 to support weighted GPAs)
  - Mutual exclusivity of is_current and end_date fields
- Education entries are ordered by rank for proper display ordering
- Added cross-user access prevention to ensure users can only access their own education records

**Closes:** #286 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

Added comprehensive test suite in `test_education_service.py` covering:

- [x] Create, read, update, delete with proper error handling for nonexistent users/entries
- [x] Data validation for required fields, date relationships, GPA range, and `is_current`/`end_date` mutual exclusivity
- [x]Cross-user access prevention ensures users can only access their own education records

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

### CODE REVIEW

File: tests/test_education_service.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+def test_gpa_validation(test_user):
+    """Test GPA validation on create and update."""
+    # Invalid GPA rejected on create
+    assert create_education(test_user, {"institution": "MIT", "degree": "BS", "gpa": -0.5}) is None
+    assert create_education(test_user, {"institution": "MIT", "degree": "BS", "gpa": 5.5}) is None
+
+    # Valid GPA accepted (standard 4.0, weighted 5.0, zero edge case)
+    result = create_education(test_user, {"institution": "MIT", "degree": "BS", "gpa": 3.75})
+    assert result["gpa"] == 3.75
```
Comment (Ron-it) [inline tests/test_education_service.py]: the comment mentions "zero edge case" and "weighted 5.0" but these boundaries don't seem to be actually tested here?

could you modify this to add boundary tests (0.0, 4.0, 5.0)? or just update the comment

**Comment by ojusharma** [comment]
> lgtm! thanks for addressing the comments
> 
> just wanted to confirm if the endpoints for these will be added in a separate pr?

yup!! The pr was 400ish lines already. Will add endpoints next week.
